### ISSUE TITLE
When comic name ends with '.' script will fail saving single files on Windows (can't create directory ending with dot)

### ISSUE THREAD

**Issue by kesipyc** [Cortys/comixology-backup#30]
Like if comic name ends with "R.I.P.".

master branch.
I could fix it changing common.js downloadFile.
Something like "name.replace(/\.\//, '/')".

Proper fix should probably be a little different.

**Comment by Cortys** [comment]
Thanks for the report. 

Do you have an idea how this should be handled? Should the dot be removed or replaced by something else?

**Comment by kesipyc** [comment]
My guess is that something treats this as dot separating name from extension. Then extension is empty and dot is ignored. So maybe if last dot was replaced with two dots only for writing it would work. But it would be confusing to have it at the end.
I'd just remove it (well I did in example).

My main concern was that there are probably more cases similar like that.
Probably they are as rare so it's enough to just react when someone finds it.

**Comment by sgbeal** [comment]
FWIW, as a minor statistical anecdote: i've backed up over 1200 books and have not yet had this problem. i.e. it's probably quite rare.

This is apparently a Windows-specific problem, but is likely a problem of the filesystem, and not Windows itself. Since many people use FAT filesystems on other OSes (e.g. most SD/USB disks use some form of FAT), it the trailing period should probably be changed for all platforms. (Either removed or replaced with an underscore, or similar.)

**Comment by Cortys** [comment]
I started looking into this issue and as @kesipyc said, naming rules on Windows are quite a mess...
https://msdn.microsoft.com/en-us/library/aa365247

Fixing issues only when they are reported is probably the easiest way to handle this.

**Comment by sgbeal** [comment]
Sounds fair to me, but keep in mind that the limit is not really Windows, but the FAT family of filesystems. Almost all external USB drives come pre-formatted as VFAT (or one of the FAT variants). Windows might  (i'm not certain) place further limits on names.

That said, i store my backups on 2 separate USB drives, both FAT-formatted, and have not yet seen any problems in this regard.

**Comment by Cortys** [comment]
I just finished a fix for this issue and (in theory) all other file name related ones.

All file names are now sanitized to be compliant to Microsofts official file name requirements, as well as the *nix requirements. FAT LFN compliance should be guaranteed as well (I haven't tested that though).

As I looked at the specs I recognized, that spaces are actually supported in every major FS and OS, so I decided to keep them in as they are. If you are no friends of spaces in file names, this of course can be configured in the options (replace with `.`, `_`, `-` or remove spaces).

https://github.com/Cortys/comixology-backup/tree/2.1.1

Does this work for you?
